cernopendata_client/downloader.py

```diff
@@ -179,41 +263,63 @@ def downloader_file_checker(file_location, file_dest):
     return False
 
 
-def download_single_file(path=None, file_location=None, protocol=None):
-    """Download single file."""
+def download_single_file(
+    path=None, file_location=None, protocol=None, download_engine=None
+):
+    """Download a single file.
+
+    :param path: Directory where file is downloaded
+    :param file_location: Remote location of a file
+    :param protocol: Protocol to be used for downloading a file
+    :param download_engine: Library to be used in downloading files
+    :type path: str
+    :type file_location: str
+    :type protocol: str
+    :type download_engine: str
+
+    :return: None
+    :rtype: None
+    """
     file_name = file_location.split("/")[-1]
     file_dest = path + "/" + file_name
     if protocol in ["http", "https"]:
-        if pycurl_available:
-            file_status = downloader_file_checker(file_location, file_dest)
-            if file_status:
-                file_size_offline = os.path.getsize(file_dest)
+        download_engine_map = {
+            "requests": requests_available,
+            "pycurl": pycurl_available,
+        }
+        if download_engine:
+            if not download_engine_map.get(download_engine):
                 display_message(
-                    msg_type="note",
-                    msg="File {} is incomplete. Resuming download.".format(
-                        file_name,
+                    msg_type="error",
+                    msg="{} is not installed on system. Please install it".format(
```

Comment:
Note that there is a traceback coming after the message. We should exit without showing traceback. Also, please put a full stop after the error message.

```console
$ cernopendata-client download-files --recid 5500 --download-engine=pycurl
==> Downloading file 1 of 11
==> ERROR: pycurl is not installed on system. Please install it
Traceback (most recent call last):
  File ".../cernopendata-client/bin/cernopendata-client", line 8, in <module>
    sys.exit(cernopendata_client())
  File ".../cernopendata-client/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File ".../cernopendata-client/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File ".../cernopendata-client/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File ".../cernopendata-client/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File ".../cernopendata-client/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File ".../cernopendata-client/lib/python3.9/site-packages/cernopendata_client/cli.py", line 348, in download_files
    download_single_file(
  File ".../cernopendata-client/lib/python3.9/site-packages/cernopendata_client/downloader.py", line 322, in download_single_file
    downloader.file_downloader()
  File ".../cernopendata-client/lib/python3.9/site-packages/cernopendata_client/downloader.py", line 132, in file_downloader
    c = pycurl.Curl()
NameError: name 'pycurl' is not defined
```

Reply:
Another observation on the same track: is protocol option being checked against download engine option?

```console
$ cernopendata-client download-files --recid 5500 --protocol xrootd --download-engine requests
==> Downloading file 1 of 11
```

One would expect an error message saying that protocol xrootd and engine requests are not compatible. 

Please add some checks and test cases for that.